contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2509,3 +2510,170 @@ get_owner_of_schema(const char *schema)
 
 	return result;
 }
+
+/*
+ * get_proc_namespace_oid
+ * Find namespace oid of a procedure based on proc name.
+ */
+static Oid
+get_proc_namespace_oid(char **proc_name, char *curr_db)
+{
+	char *physical_sch_name;
+	char *db_name;
+	char *schema_name;
+	char **splited_object_name;
+	Oid obj_schema_oid = InvalidOid;
+
+	/* Resolve the three part name. */
+	splited_object_name = split_object_name(*proc_name);
+	db_name = splited_object_name[1];
+	schema_name = splited_object_name[2];
+	*proc_name = splited_object_name[3];
+
+	/* downcase identifier if needed */
+	if (pltsql_case_insensitive_identifiers)
+	{
+		db_name = downcase_identifier(db_name, strlen(db_name), false, false);
+		schema_name = downcase_identifier(schema_name, strlen(schema_name), false, false);
+		*proc_name = downcase_identifier(*proc_name, strlen(*proc_name), false, false);
+		for (int j = 0; j < 4; j++)
+			pfree(splited_object_name[j]);
+	}
+	else
```

Comment:
Seems like same code has been reused multiple places, possibilities of creating a utility?